src/ert/scheduler/lsf_driver.py

```diff
@@ -107,19 +107,26 @@ async def submit(
             stderr=asyncio.subprocess.PIPE,
         )
         stdout, stderr = await process.communicate()
-        try:
-            job_id = (
-                stdout.decode("utf-8")
-                .strip()
-                .replace("<", "")
-                .replace(">", "")
-                .split()[1]
-            )
-        except IndexError as err:
+        if process.returncode:
             logger.error(
-                f"Command \"{' '.join(bsub_with_args)}\" failed with error message: {stderr.decode()}"
+                f"Command \"{' '.join(bsub_with_args)}\" failed with "
+                f"returncode {process.returncode} and error message: {stderr.decode() or '<empty>'}"
             )
-            raise RuntimeError from err
+            raise RuntimeError(stderr.decode())
+
+        stdout_decoded = stdout.decode()
+        if not (
+            stdout_decoded.startswith("Job ")
+            and "is submitted to " in stdout_decoded
+            and "queue" in stdout_decoded
+        ):
+            raise RuntimeError(f"Could not understand {stdout_decoded} from bsub")
+
+        job_id = (
+            stdout.decode("utf-8").strip().replace("<", "").replace(">", "").split()[1]
+        )
+        if not (job_id.isdigit() and int(job_id) > 0):
+            raise RuntimeError(f"job_id {job_id} is not a positive number")
```

Comment:
`import re`

```suggestion
        match = re.search(b"Job <([0-9]+)> is submitted to .+ queue")
        if match is None:
            raise RuntimeError(f"Could not understand {repr(stdout.decode())} from bsub")

        job_id = int(match[1])
```

If you want to explicitly check for a negative value, you can the regex to `b"Job <(-?[0-9]+])> is submitted to .+ queue"` and then check whether `int(match[1])` is negative.

Reply:
Great. Not sure why I wanted to avoid regexp in the first place, this is neater. Skipping treating negative number specifically, and for simplicity I will allow a zero job id. I have no reference to say that a zero is not permitted.